src/package-linker.js

```diff
@@ -368,16 +368,20 @@ export default class PackageLinker {
 
     for (const name in peerDeps) {
       const range = peerDeps[name];
-      const patterns = this.resolver.patternsByPackage[name] || [];
-      const foundPattern = patterns.find(pattern => {
-        const resolvedPattern = this.resolver.getResolvedPattern(pattern);
-        return resolvedPattern ? this._satisfiesPeerDependency(range, resolvedPattern.version) : false;
+      const pkgs = this.resolver.getAllInfoForPackageName(name);
+      let foundPattern;
+      pkgs.find(pkg => {
```

Comment:
I think what you intend to do is this:
```
const found = pkgs.find(pkg => {
    const {root, version, patterns} = pkg._reference || {};
    return root && this._satisfiesPeerDependency(range, version) && patterns;
});

if (found) {
    ref.addDependencies(found._reference.patterns);
}
```

Modifying the outside scope in iteration callbacks are a bit dangerous and quite confusing to the reader :(